examples/fetch_ic/test/tests.ts

```diff
@@ -0,0 +1,318 @@
+// TODO allow running these tests with and without the authorization header
+// TODO these seem like really good tests for the server and normal fetch too
+// TODO consider how we'll do the axios tests as well
+
+import * as dns from 'node:dns';
+dns.setDefaultResultOrder('ipv4first');
+
+import { Principal } from '@dfinity/principal'; // TODO get rid of this dependency if we installed it
+import { Test, getCanisterOrigin } from 'azle/test';
+import puppeteer, { Browser, Page } from 'puppeteer';
+
+export function getTests(canisterName: string): Test[] {
+    const origin = getCanisterOrigin(canisterName);
```

Comment:
Do we need test for:
- Make sure adding unsupported options doesn't break anything
- PUT
- DELETE
- PATCH
- GET and/or HEAD with a body defined
- FormData
- ReadableStream